solutions/kra/lab3/kern/pmap.c

```diff
@@ -100,7 +100,17 @@ boot_alloc(uint32_t n)
 	//
 	// LAB 2: Your code here.
 
-	return NULL;
+	if (n == 0) {
+		return nextfree;
+	}
+
+	if((uint32_t)nextfree - KERNBASE >= 0xffffffff) {
```

Comment:
Please explain/eliminate magic numbers. What is the size of KERNBASE? What the meaning for >= ?

Reply:
The statement was corrected and simplified.